sqlmesh/core/context.py

```diff
@@ -1268,7 +1268,7 @@ def table_diff(
         self,
         source: str,
         target: str,
-        on: t.List[str] | exp.Condition | None = None,
+        on: t.List[str | exp.Identifier] | exp.Condition | None = None,
```

Comment:
This was to account for quoted identifiers, since when I ran a few tests with quoted identifiers in the grain, like lowercase "id" in Snowflake, these were not handled

The information if they're quoted or not is saved here: [L1326](https://github.com/TobikoData/sqlmesh/blob/ad8193a6f30cdcf1656af0a33ff678649e84be47/sqlmesh/core/context.py#L1326)

Reply:
why don't you just have quoted strings?

Reply:
ok I'll try this instead